src/api/getEvents.ts

```diff
@@ -118,17 +119,44 @@ interface EndpointReturn
 }
 
 // used to get the lattitude from an address 
-async function getlat(location: string): Promise<number>
+// async function getLattitudeAndLongitude(location: geocoder.Query): Promise<geocoder.Location>
+async function getLattitudeAndLongitude(address: string, city: string, state: string, zip: string): Promise<Coordinates>
 {
-	let lattitude: number = (await geoCoder.geocode('location')).lattitude;
-	return lattitude;
-}
+	let coordinates: Coordinates = {
+		lattitude: 0,
+		longitude: 0
+	}
 
-// used to get the longitude from an address
-async function getlong(location: string): Promise<number>
-{
-	let longitude: number = (await geoCoder.geocode('location')).longitude;
-	return longitude;
+	if (process.env.TOMTOM_KEY === undefined)
+	{
+		console.warn("TOM TOM api key not set");
+		return coordinates;
+	}
+
+	const options: geocoder.GenericOptions = {
+		provider: "tomtom",
+		apiKey: process.env.TOMTOM_KEY
+	};
+
+	const geocoder: geocoder.Geocoder = node_geocoder(options);
+
+	let locations: Array<Entry> = await geocoder.geocode(address + " " + city + " " + state + " " + zip);
+
+	// parse the data of the first match
+	if (locations.length > 0)
+	{
+		if (locations[0].latitude !== undefined)
+		{
+			coordinates.lattitude = locations[0].latitude;
+		}
+
+		if (locations[0].longitude !== undefined)
+		{
+			coordinates.longitude = locations[0].longitude;
+		}
+	}

```

Comment:
Probably safe to do something like

`locations.length > 0 ? locations : undefined`

or 

`const { latitude, longitude } = locations`

`latitude && longitude ? { lng: longitude, lat: latitude} : undefined`

Reply:
Implemented safer practice:
```typescript
	// parse the data of the first match
	if ((locations !== undefined) && (locations.length > 0))
	{
		const { latitude, longitude } = locations[0];

		coordinates.latitude = latitude;
		coordinates.longitude = longitude;
	}
```